pyro/infer/tracegraph_elbo.py

```diff
@@ -155,7 +155,8 @@ def _compute_elbo_non_reparam(guide_trace, guide_vec_md_nodes,  #
         if log_pdf_key == 'log_pdf':
             score_function_term = score_function_term.sum()
         if use_nn_baseline or use_decaying_avg_baseline or use_baseline_value:
-            if downstream_cost.size() != baseline.size():
+            if (downstream_cost.dim() == 0 and baseline.dim() > 1) or \
```

Comment:
@fritzo, @martinjankowiak - This works fine now, but please feel free to modify this logic later. The issue is that `downstream_cost` can be a dim 0 tensor now whereas the baseline will always have dim > 0 as the output from a nn module can't be dim 0. I tried a couple of other approaches, but this seemed like the cleanest way to get all the tests working.